Solves Issue #32750 - torch.prod now works fine with FP16 Input Tensor and FP32 Output Tensor

krshrimali opened the pull request:
This PR solves Issue #32750. Changes function `prod_kernel_impl` to use `acc_t` argument instead of `scalar_t` (which caused the garbage output for FP16 input and FP32 output tensor type).

krshrimali committed:
torch.prod works fine now with fp16 input and fp32 output tensor

krshrimali committed:
torch.prod works fine now with fp16 input and fp32 output tensor

zasdfgbnm committed:
Deduplication of type casting codes (#32730)

Summary:
These codes are implemented twice at different places by different people, we should merge them together.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32730

Differential Revision: D19622023

Pulled By: ezyang

fbshipit-source-id: a9cbda31428b335bf28a7e4050f51f58e787b94f

Commit:
More code fakefp16 mapping unification

Summary: ATT

Reviewed By: amylittleyang

Differential Revision: D19597036

fbshipit-source-id: deed61945884fb4b01d058f3c72c75f5a937a41c

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit dc0e6f35:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=dc0e6f35f2f0dc05aae73967b9bee2a8006c18bd).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4370257) pytorch_windows_test2 (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=92703656) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=92703656)) :snowflake:
<details>
<summary>
<code>AssertionError: tensor(1.1904) not less than or equal to 0.01 :</code>
</summary>

```
---------------------------------------------------------------------- 
Traceback (most recent call last): 
  File "C:\Users\circleci\project\build\win_tmp\build\torch\testing\_internal\common_device_type.py", line 197, in instantiated_test 
    result = test(self, device_arg, dtype) 
  File "test_torch.py", line 14549, in fn 
    self.assertEqual(cpu_result, device_result, prec=precision) 
  File "C:\Users\circleci\project\build\win_tmp\build\torch\testing\_internal\common_utils.py", line 862, in assertEqual 
    assertTensorsEqual(x, y) 
  File "C:\Users\circleci\project\build\win_tmp\build\torch\testing\_internal\common_utils.py", line 832, in assertTensorsEqual 
    self.assertLessEqual(max_err, prec, message) 
AssertionError: tensor(1.1904) not less than or equal to 0.01 :  
 
---------------------------------------------------------------------- 
Ran 2911 tests in 45.959s 
 
FAILED (failures=1, errors=741, skipped=43) 
Traceback (most recent call last): 
  File "run_test.py", line 457, in <module> 
    main() 
  File "run_test.py", line 450, in main 
    raise RuntimeError(message) 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

pbelevich committed:
Custom RNG DispatchKey (#32325)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32325

The purpose of this PR is to enable PyTorch dispatching on `at::Generator*` parameters and demonstrate how it can be used in cpp extensions to implement custom RNG.
1. `CustomRNGKeyId` value added to DispatchKey enum and `DispatchKeySet key_set_` added to `at::Generator`
2. The overloaded `operator()(at::Generator* gen)` added to MultiDispatchKeySet.
3. The existing CPUGenerator and CUDAGenerator class are supplied with CPUTensorId and CUDATensorId dispatch keys
4. The implementation of CPU's `cauchy_kernel`(as an example, because it's already moved to ATen) was templatized and moved to `ATen/native/cpu/DistributionTemplates.h` to make it available for cpp extensions
5. Minor CMake changes to make native/cpu tensors available for cpp extensions
6. RegisterCustomRNG test that demonstrates how CustomCPUGenerator class can be implemented and how custom_rng_cauchy_ native function can be registered to handle Tensor::cauchy_ calls.

Test Plan: Imported from OSS

Differential Revision: D19604558

Pulled By: pbelevich

fbshipit-source-id: 2619f14076cee5742094a0be832d8530bba72728

yf225 commented:
@nairbv Would you be interested in reviewing it?

dylanbespalko committed:
Improved speed of frobenous norm for non-complex dtype (#30871)

Summary:
In-tree changes to pytorch to support complex numbers are being submitted here.
Out-of-tree support for CUDA complex numbers is here: [pytorch-cuda-strided-complex extension](https://gitlab.com/pytorch-complex/pytorch-cuda-strided-complex)

Changes:
[x] Fixed performance issue raise in https://github.com/pytorch/pytorch/issues/30704 so that non-complex numbers do not call `conj()` and `real()`.
[x] Fixed tensor_to_numpy() conversion likely broken by a `checkBackend()` in https://github.com/pytorch/pytorch/issues/27064.
[x] Fixed some ReduceOps and TensorCompare Ops that recently added a `checkBackend()`.
    - `checkBackend()` is replaced with a device type check and a layout check.
    - This ensures the ComplexCPU Type ID is supported.
[x] Added AVX support for complex `exp()`, as requested in https://github.com/pytorch/pytorch/issues/755
Pull Request resolved: https://github.com/pytorch/pytorch/pull/30871

Differential Revision: D19200726

Pulled By: ezyang

fbshipit-source-id: d7e1be0b0a89c5d6e5f4a68ce5fcd2adc5b88277

IvanKobzarev committed:
[android] fbjni DoNotStrip annotation for oss native methods (#32567)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32567

As a first change to support proguard.
even if these methods could be not called from java, on jni level we register them and this registration will fail if methods are stripped.

Adding DoNotStrip to all native methods that are registered in OSS.

After integration of consumerProguardFiles in fbjni that prevents stripping by proguard DoNotStrip it will fix errors with proguard on.

Test Plan: Imported from OSS

Differential Revision: D19624684

Pulled By: IvanKobzarev

fbshipit-source-id: cd7d9153e9f8faf31c99583cede4adbf06bab507

zhaojuanmao committed:
clean up GIL usuage (#32748)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32748

This is to follow up PR #30630, we need to have GIL when calling jit::toPyObject(), for some binded functions need to be taged with GIL release if underneath C++ codes requires GIL. so
1. pyRef::to_here() and pyRef::local_value() added GIL
2. pyRef::pickle and pyRef::unpickle() added GIL release tag
3. in request_callback_impl, also added GIL as needed
4. for typeParser, use cached jitCompilationUnit_, also clean it up in cleanUp() function
ghstack-source-id: 97373011

Test Plan: unit test

Differential Revision: D19612337

fbshipit-source-id: 4d09f9b52ba626545ae7d31fea6b671301ed3890

Commit:
Randomize xla port (#32718)

Summary:
fixes https://github.com/pytorch/pytorch/issues/30717
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32718

Differential Revision: D19607998

Pulled By: ailzhang

fbshipit-source-id: 81ba9c7c71988a64cdc8fa5500967509657438fe

deepali-c committed:
Fix TensorProtosDBInput AttributeError (#32274)

Summary:
https://github.com/pytorch/pytorch/issues/6794
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32274

Differential Revision: D19621889

Pulled By: ezyang

fbshipit-source-id: 1bdd042b6421a2798c7f1e9030dfc6dfc1246989

nairbv commented on aten/src/ATen/native/cuda/ReduceOpsKernel.cu:
```diff
@@ -41,7 +41,7 @@ void std_var_kernel_impl<at::Half>(TensorIterator& iter, bool unbiased, bool tak
 template <typename scalar_t, typename acc_t=scalar_t>
 void prod_kernel_impl(TensorIterator& iter) {
-  gpu_reduce_kernel<scalar_t, scalar_t>(iter, func_wrapper<scalar_t> ([]GPU_LAMBDA(acc_t a, acc_t b) -> acc_t {
+  gpu_reduce_kernel<scalar_t, acc_t>(iter, func_wrapper<scalar_t> ([]GPU_LAMBDA(acc_t a, acc_t b) -> acc_t {
```
we should have a test

krshrimali replied:
Sure, @nairbv. I might have to do a few more modifications to the `prod_kernel_impl` as well. And also write a unit test, & commit. Thanks!

ezyang committed:
Make VC++ version a parametrizable option for Windows CI. (#32043)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32043

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19621910

Pulled By: ezyang

fbshipit-source-id: dce00a56ff679548fd9f467661c3c54c71a3dd4e

kurtamohler committed:
Port BCELoss to ATen to increase accuracy (#31365)

Summary:
Fixes issue https://github.com/pytorch/pytorch/issues/24933
Pull Request resolved: https://github.com/pytorch/pytorch/pull/31365

Differential Revision: D19557712

Pulled By: ezyang

fbshipit-source-id: 3ae78c949b2f6c21b294d986d28e09daa9b0c526

ezyang committed:
Improve documentation in dispatcher; remove unnecessary optional (#32533)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32533

Applies renames based on comments in #32439.  I also updated some
other documentation and variable names while I was at it.

Fixes #32435.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19579854

Pulled By: ezyang

fbshipit-source-id: 85021a92a2a84501f49ee5c16318f81f5df64f8d

ezyang committed:
Add missing C10_API to dispatch key TLS setter/getters (#32557)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32557

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19579853

Pulled By: ezyang

fbshipit-source-id: 45f83a7a5ead0344e4c13526abb5fafdedaed4a4

ezyang committed:
Delete copy/move constructors on these RAII guards. (#32727)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32727

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19621858

Pulled By: ezyang

fbshipit-source-id: 5112c849252478d8249de4f8c8c5a2d6caf60672

ezyang committed:
Refreshing numel on a stride update is pointless. (#32116)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32116

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19579875

Pulled By: ezyang

fbshipit-source-id: 00393c9dc101967c79231bfae36b23b7b80135fb

ezyang committed:
Delete resize_dim() (#32114)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32114

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19579876

Pulled By: ezyang

fbshipit-source-id: d09a231ba891403a06eae0c2203e0ad7dd6d3a12

ezyang committed:
Expunge last two sites of resize_dim (#32112)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32112

It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Test Plan: Imported from OSS

Differential Revision: D19579874

Pulled By: ezyang

fbshipit-source-id: e40efbf94e128fd81421b227b76dd9c9c0256d96

ezyang committed:
Revert D19597036: More code fakefp16 mapping unification

Test Plan: revert-hammer

Differential Revision:
D19597036

Original commit changeset: deed61945884

fbshipit-source-id: c057e57810a99464aefb00b645613ecd6a7c5533

zasdfgbnm committed:
Fix ivalue_inl.h:353:29: warning: comparison of unsigned expression >= 0 is always true (#32778)

Summary:
`slot` is unsigned integer which is `always >= 0`
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32778

Differential Revision: D19625789

Pulled By: ngimel

fbshipit-source-id: c92c35c65d4372be934283e87aeba99e9e0ef353

pbelevich committed:
Remove -Werror from test/cpp_extensions/setup.py (#32704)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32704

-Werror is too aggressive check for test cpp extensions because it fails even on deprecation warnings which is are included from core codebase.

Fixes #32136

Test Plan: Imported from OSS

Differential Revision: D19620190

Pulled By: pbelevich

fbshipit-source-id: 0e91566eb5de853559bb59e68a02b0bb15e7341b

Commit:
[quantization] Remove incorrect fp16 dynamic linear/relu op

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/32774

Test Plan: Imported from OSS

Differential Revision: D19624471

Pulled By: jamesr66a

fbshipit-source-id: eb6cb11fabf2ddd5edf345aff35b86b83c3af94c

Enealor committed:
Logspace fixes (#32744)

Summary:
Reopening of PR https://github.com/pytorch/pytorch/issues/32631 with `viable/strict` base for testing
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32744

Differential Revision: D19626090

Pulled By: ngimel

fbshipit-source-id: ed0fc759198ee2edc23afdcb1e190a11d70ec4c8

jianyuh committed:
[pytorch][embeddingbag_8bit] Add include_last_offset option to Fused 8bit EmbeddingBag and parallelize the op (#32683)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32683

Pull Request resolved: https://github.com/pytorch/glow/pull/4079

Similar to D17768404, we changed the EmbeddingBag operator for 8-bit fused version to add the option to include the last offset and parallelize the op.
ghstack-source-id: 97404645

Test Plan:
To generate the AVX2 code (`embedding_lookup_fused_8bit_rowwise_idx_avx2.cc`):
```
python hp_emblookup_codegen.py --fused --use-offsets
```

To test the correctness:

```
buck test //caffe2/torch/fb/sparsenn:test -- test_embedding_bag_byte_rowwise_offsets  --print-passing-details
```

Reviewed By: yinghai

Differential Revision: D19592761

fbshipit-source-id: f009d675ea3f2228f62e9f86b7ccb94700a0dfe0

Commit:
enable empty batch for all flavor of convolutions (#32709)

Summary:
resubmitting https://github.com/pytorch/pytorch/issues/32612 after a merge gone wrong. Enables convolution with an empty batch or number of channels for all flavors of convolution (grouped convolution, convTranspose). Would make https://github.com/pytorch/pytorch/issues/31658 unnecessary. Also returns zero gradients for the parameters, that's necessary for correct DDP operation.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32709

Differential Revision: D19627968

Pulled By: ngimel

fbshipit-source-id: 7359759bd05ff0df0eb658cac55651c607f1b59f

suo committed:
[jit] fix the NoneType param/buffer hack (#32745)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32745

Some parameters (like `bias` in conv) are optional. To achieve this
previously, you had to add `bias` as a constant, which would invoke some
pretty weird behavior in the frontend, summarized as:
```
if bias is not None:
  add it as a parameter normally
else: # bias is None
  add it as a constant with the value None
```

There are several things bad about this:
1. Bias is not a constant. Marking it `__constants__` is confusing.
2. It basically relies on an implementation detail (the frontend
processes parameters before constants) to work.

Okay, whatever. I don't even know why we did this originally, but
getting rid of it doesn't break anything, so I assume improved NoneType
refinement has made this a non-issue.

Note on perf: this will make no difference; if bias was `None` it's still
folded out today, if bias is a Tensor it would be added as a parameter
both before and after this change

Test Plan: Imported from OSS

Differential Revision: D19628634

Pulled By: suo

fbshipit-source-id: d9128a09c5d096b938fcf567b8c23b09ac9ab37f

krshrimali committed:
Replicates sum_kernel_cuda and sum_kernel_impl, adds out_t arg

krshrimali committed:
Replicates sum_kernel_cuda and sum_kernel_impl, adds out_t arg

mingzhe09088 committed:
Resolve rendezvous race condition

Summary:
When running the ctr_mbl_feed, we've encountered hang issue related to the rendezvous handshake based on zeus. It was mitigated by this diff https://our.intern.facebook.com/intern/diff/D19167151/.

This diff resolves the race condition by adding a reference to the rendezvous handler.

Test Plan: x7340282797

Reviewed By: yifuwang

Differential Revision: D19627293

fbshipit-source-id: 560af289db8ef6cf8d6f101f95ec27d5a361fd04

krshrimali committed:
Test added for test_prod_gpu, under review, passes with changes successfully

krshrimali committed:
Test added for test_prod_gpu, under review, passes with changes successfully

mvsampath committed:
[PyTorch][TorchScript] Add support for join on List of strings in TorchScript

Summary: Add support for join on List of strings in TorchScript.

Test Plan:
(pytorch) smummadi@smummadi-mbp pytorch % python test/test_jit_string.py
Fail to import hypothesis in common_utils, tests are not derandomized
.
----------------------------------------------------------------------
Ran 1 test in 1.090s

OK

Differential Revision: D19611800

fbshipit-source-id: cef66356abc14dfd100a806d25dd1a8bc9af0a11

Commit:
Fixed access to element in size tensor for scripting (#32652)

Summary:
when using scripting, there was an error in attempting to access a
specific element from within the size tensor.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32652

Reviewed By: hl475

Differential Revision: D19610726

Pulled By: houseroad

fbshipit-source-id: bca49927bbe71dbe7e7d7edf301908fe79e089b5

krshrimali committed:
modified test for torch.prod (only CUDA), uses all dtype combinations

krshrimali committed:
modified test for torch.prod (only CUDA), uses all dtype combinations

krshrimali committed:
use torch.tensor.dtype instead of torch.tensor.type()

krshrimali committed:
use torch.tensor.dtype instead of torch.tensor.type()

supriyar committed:
Add support for Dynamic LSTM quantization on Mobile (#32757)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32757

This PR updates the main quantize_dynamic API to use QNNPACK backend for mobile

Test Plan:
python test/test_quantization.py PostTrainingDynamicQuantTest.test_quantized_rnn

Imported from OSS

Differential Revision: D19632220

fbshipit-source-id: b4c51485c281d088524101b97c84dd806438b597

Commit:
[pytorch] Minor: avoid acquiring GIL twice in PyRRef::localValue() (#32785)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32785

Add PythonRpcHandler::handleExceptionWithGIL() so that in PyRRef::localValue(),
we don't need to release the GIL and re-acquire the following line.
ghstack-source-id: 97418465

Test Plan: existing test coverage

Differential Revision: D19626195

fbshipit-source-id: db694d04b078811f819626789e1e86f1b35adb5b

Commit:
Fix/simplify alias annotation handling in op codegen. (#32574)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32574

Previously, we ignored alias annotations when deriving argument mutability
and instead recognized particular signature patterns (in-place, out variant)
and assigned mutability accordingly. Op signatures that didn't fit these
patterns would error (e.g. see #30526, which this fixes).

No change in the generated binding code.

Code changes:
1. in function_wrapper.py, fix the mutability derivation logic used when creating an argument's c++ type property. Note that we temporarily need to trap a special case and apply the old logic, see code comment for details.

2. in gen_jit_dispatch.py, update logic that assumed only one mutable Tensor argument per declaration. Happily this mostly was accomplished by bypassing some now-redundant signature regeneration machinery. Another special case here requires that we keep the old machinery around temporarily.

Test Plan: Imported from OSS

Differential Revision: D19564875

Pulled By: bhosmer

fbshipit-source-id: 5637a9672923676d408c9586f3420bcc0028471a

Commit:
Some work on eager op binding codegen (gen_python_functions.py) (#29986)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/29986

Previously in addition to generating a python binding for each op,
we would generate an almost-trivial helper for each overload.
This PR eliminates the helpers, simplifying codegen logic a bit and
reducing the source-level indirection by a step.
Perf should be unchanged.

codegen diff: https://github.com/bhosmer/scratch/commit/1f2f07fb605e782cf7fdfb7d5eb33050eb65a6b4

Note: in the interests of keeping the diff contained, there's only
some light cleanup here beyond what's necessary for the codegen changes.
Plan is to do some more substantial refactoring in followup PRs that
leave generated code unchanged.

Test Plan: Imported from OSS

Differential Revision: D18567980

Pulled By: bhosmer

fbshipit-source-id: eb9a81babb4489abd470842757af45580d4c9906

xush6528 committed:
Reduce RPC branches for Python/BuiltinOp/TorchScript (#32689)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32689

As described in https://github.com/pytorch/pytorch/issues/32565
ghstack-source-id: 97440343

Test Plan:
```
buck test mode/dev-nosan //caffe2/test/distributed/rpc:rpc_fork -- test_script_functions_not_supported

buck build mode/dev-nosan //caffe2/test/distributed/rpc:rpc_fork

buck-out/gen/caffe2/test/distributed/rpc/rpc_fork\#binary.par -r test_script_functions_not_supported
```

```
buck test mode/dev-nosan //caffe2/test/distributed/rpc:dist_autograd_fork

buck build mode/dev-nosan //caffe2/test/distributed/rpc:dist_autograd_fork

buck-out/gen/caffe2/test/distributed/rpc/dist_autograd_fork\#binary.par -r test_backward_simple_script_call
```

Differential Revision: D5721814

fbshipit-source-id: 9079e81764be1e7c7b85dd72a18c76f3ecfd2547

ezyang committed:
Revert D19611800: [PyTorch][TorchScript] Add support for join on List of strings in TorchScript

Test Plan: revert-hammer

Differential Revision:
D19611800

Original commit changeset: cef66356abc1

fbshipit-source-id: 41af9e0de83b1fb808b17255ec905e137909457d

pbelevich committed:
Removing @expectedFailureXLA from test_nll_loss_empty_tensor_reduction_mean (#32701)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32701

Because it's disabled in XLA(https://github.com/pytorch/xla/pull/1563)
Discussed in https://github.com/pytorch/xla/issues/1539

Test Plan: Imported from OSS

Differential Revision: D19633349

Pulled By: pbelevich

fbshipit-source-id: b9a81c976a96b325356ff210ff838dfcd5352db7

albanD committed:
Re-enable non-deterministic autograd tests

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/32793

Test Plan: Imported from OSS

Differential Revision: D19634632

Pulled By: albanD

fbshipit-source-id: 9dda29536c2ed4afb81ecbea471ba615241bbac2

Commit:
Fix backcompat for linear_relu_dynamic_fp16 (#32803)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32803

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32803 Fix backcompat for linear_relu_dynamic_fp16**

Test Plan: Imported from OSS

Reviewed By: jamesr66a

Differential Revision: D19642281

Pulled By: albanD

fbshipit-source-id: 3b6ae4dd81bf8a70dd81ccbb02fffd7653bbd08c

peterjc123 committed:
Make python version a parameterizable option for Windows CI.

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/32823

Differential Revision: D19642347

Pulled By: ezyang

fbshipit-source-id: a4d461aa29a06bb7f5e5d359a2df2c90e9a4fd41

krshrimali committed:
Merge branch 'master' of https://github.com/krshrimali/pytorch

krshrimali commented:
Closing this PR, as I seemed to have messed up things while trying to re-base.
